pkg/tools/gen/genkcl.go

```diff
@@ -0,0 +1,146 @@
+package gen
+
+import (
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+)
+
+// TODO: is need to parse from Comment. Now get kcl schema from Tag
+type GenKclOptions struct {
+	KclPackage       string
+	ParseFromComment bool
+	ParseFromTag     bool
+}
+
+type kclGenerator struct {
+	opts *GenKclOptions
+}
+
+func GenKcl(w io.Writer, filename string, src interface{}, opts *GenKclOptions) error {
+	return newKclGenerator(opts).GenSchema(w, filename, src)
+}
+
+func newKclGenerator(opts *GenKclOptions) *kclGenerator {
+	if opts == nil {
+		opts = new(GenKclOptions)
+	}
+	return &kclGenerator{
+		opts: opts,
+	}
+}
+
+func (k *kclGenerator) GenSchema(w io.Writer, filename string, src interface{}) error {
+	fmt.Fprintln(w)
+	goStructs := ParseGoSourceCode(filename, src)
+	for _, goStruct := range goStructs {
+		fmt.Fprintf(w, "schema %s:\n", goStruct.Name)
+		if goStruct.StructComment != "" {
+			fmt.Fprintf(w, "\"\"\"%s\"\"\"\n", goStruct.StructComment)
+		}
+		for _, field := range goStruct.Fields {
+			kclFieldName, kclFieldType := k.parserGoStructFieldTag(field.FieldTag)
+			fmt.Fprintf(w, "    %s: %s\n", kclFieldName, kclFieldType)
+		}
+		fmt.Fprintf(w, "\n")
+	}
+	return nil
+}
+
+func (k *kclGenerator) parserGoStructFieldTag(tag string) (string, string) {
```

Comment:
Will this function handle error returns for illegal tag strings? Such as
```go
func (k *kclGenerator) parserGoStructFieldTag(tag string) (string, string, error)
```

Reply:
ok. I will handle the error returns.